src/utils/odata.ts

```diff
@@ -17,28 +17,34 @@ export interface GraphResponseError {
   }
 }
 
-export const odata = {
-  async getAllItems<T>(url: string, metadata?: 'none' | 'minimal' | 'full'): Promise<T[]> {
-    let items: T[] = [];
-
-    const requestOptions: CliRequestOptions = {
-      url: url,
-      headers: {
-        accept: `application/json;odata.metadata=${metadata ?? 'none'}`,
-        'odata-version': '4.0'
-      },
-      responseType: 'json'
-    };
-
-    const res = await request.get<ODataResponse<T>>(requestOptions);
-    items = res.value;
-
-    const nextLink = res['@odata.nextLink'] ?? res.nextLink;
-    if (nextLink) {
-      const nextPageItems = await odata.getAllItems<T>(nextLink, metadata);
-      items = items.concat(nextPageItems);
-    }
+function getAllItems<T>(url: string): Promise<T[]>;
+function getAllItems<T>(options: CliRequestOptions): Promise<T[]>;
```

Comment:
Let's order this one below the overloads with url as a first argument.

Reply:
Hi @martinlingstuyl 

If we do that way, it shows below error:
`Function implementation is missing or not immediately following the declaration`